Assets/LeapMotion/Core/Scripts/Hands/HandShaderGlobalSupport.cs

```diff
@@ -0,0 +1,323 @@
+
+            // Middle segments
+            segments[i++] = middle.bones[1].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = middle.bones[1].NextJoint.ToVector3().WithW(1);
+            segments[i++] = middle.bones[2].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = middle.bones[2].NextJoint.ToVector3().WithW(1);
+            segments[i++] = middle.bones[3].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = middle.bones[3].NextJoint.ToVector3().WithW(1);
+
+            // Ring segments
+            segments[i++] = ring.bones[1].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = ring.bones[1].NextJoint.ToVector3().WithW(1);
+            segments[i++] = ring.bones[2].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = ring.bones[2].NextJoint.ToVector3().WithW(1);
+            segments[i++] = ring.bones[3].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = ring.bones[3].NextJoint.ToVector3().WithW(1);
+
+            // Pinky segments
+            segments[i++] = pinky.bones[1].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = pinky.bones[1].NextJoint.ToVector3().WithW(1);
+            segments[i++] = pinky.bones[2].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = pinky.bones[2].NextJoint.ToVector3().WithW(1);
+            segments[i++] = pinky.bones[3].PrevJoint.ToVector3().WithW(1);
+            segments[i++] = pinky.bones[3].NextJoint.ToVector3().WithW(1);
+          }
+        }
+
+      }
+      else {
+        for (int i = 0; i < _leftFingertips.Length; i++) {
+          _leftFingertips[i] = (Vector3.one * 100000000f).WithW(1);
+        }
+        for (int i = 0; i < _leftPalmTris.Length; i++) {
+          _leftPalmTris[i] = (Vector3.one * 100000000f).WithW(1);
+        }
+        for (int i = 0; i < _leftFingerSegments.Length; i++) {
+          _leftFingerSegments[i] = (Vector3.one * 100000000f).WithW(1);
+        }
+      }
+      Shader.SetGlobalVectorArray(_leftFingertipsUniformParamId, _leftFingertips);
+      Shader.SetGlobalVectorArray(_leftPalmTrisUniformParamId, _leftPalmTris);
+      Shader.SetGlobalVectorArray(_leftFingerSegmentsUniformParamId, _leftFingerSegments);
+
+      var rightHand = provider.CurrentFrame.Hands.Query()
+                      .FirstOrDefault(h => !h.IsLeft);
+      if (rightHand != null) {
+
+        // Fingertips
+        for (int i = 0; i < NUM_FINGERS; i++) {
+          _rightFingertips[i] = rightHand.Fingers[i].TipPosition.ToVector3().WithW(1);
+        }
+
+        // Palm Tris
+        {
+          var hand = rightHand;
+          var palmTris = _rightPalmTris;
+          int i = 0;
+          
+          {
+            Finger thumb = hand.GetThumb(), index = hand.GetIndex(),
+              middle = hand.GetMiddle(), ring = hand.GetRing(), pinky = hand.GetPinky();
+
+            // Thumb Tri
+            palmTris[i++] = thumb.bones[1].NextJoint.ToVector3().WithW(1);
```

Comment:
could there be a way to remove this duplication between the left and right hands?